### ISSUE TITLE
vim_configurable starts XQuartz on startup on darwin

### ISSUE THREAD

**Issue by ejpcmac** [NixOS/nixpkgs#45883]
## Issue description

Since the last `nixpkgs-unstable` update, XQuartz is started when running `vim` in a terminal. This slows down vim startup a lot if XQuartz is not already started. vim should start without GUI.

### Steps to reproduce

1. Install vim on macOS via Nix
2. Start vim from a terminal

## Technical details

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 15.6.0, macOS 10.11.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(***): `"darwin"`
 - channels(root): `"nixpkgs-18.09pre151001.1e1221c1c15"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by ejpcmac** [comment]
@LnL7 `gvim` is not even installed on my system. I can however do `vim -g` to get the GUI and it works as expected. XQuartz should only be started when needed.

**Comment by LnL7** [comment]
Yeah but I don't know what's causing that, vim_configurable doesn't add any arguments except for `-u vimrc`

**Comment by elohmeier** [comment]
Is there already a workaround to this problem?

**Comment by bspammer** [comment]
This is still happening for me

**Comment by starcraft66** [comment]
Just started experiencing this out of the blue for unexplicable reasons. Starting vim always starts XQuartz and `vim -g` does indeed show a GUI.
